hsclient/hydroshare.py

```diff
@@ -279,12 +337,33 @@ def aggregations(self, **kwargs) -> List[BaseMetadata]:
         :return: a List of Aggregation objects matching the filter parameters
         """
         aggregations = self._aggregations
+
+        # when searching using 'file__path' or files__path' as the key, there can be only one matching aggregation
+        file_path_priority = kwargs.pop("file_path_priority", True)
+        file_path = kwargs.get("file__path", "")
```

Comment:
I don't understand why this new block is needed.  It ignores any other search parameters by default.

Reply:
Most of the aggregation data object related operations retrieve aggregation by aggregation path name only. This block makes the search bit more efficient in that case. However, the search efficiency is noticeable only when there are many aggregations in a resource or few aggregations that contain a large number of files. I have removed this aggregation path priority based search since for most of the resources it won't matter in terms of faster search.